File: src/frontend/admin/js/utils/Ajax.js
class Ajax {

    static post(url, data) {
        return fetch(url, {
            body: JSON.stringify(data),
            headers: {
                'content-type': 'application/json'
            },
            method: 'POST', // *GET, POST, PUT, DELETE, etc.
            mode: 'cors', // no-cors, cors, *same-origin
            redirect: 'follow', // manual, *follow, error
            referrer: 'no-referrer', // *client, no-referrer
        }).then(response => {
            if (response.ok) {
                return response.json()
            } else {
                throw new Error('Network response was not ok.');
            }
        })
    }

    static get(url, data) {

    }
}

module.exports = Ajax;

File: modules/frontend/src/main/scala/cz/kamenitxan/jakon/webui/Ajax.scala
package cz.kamenitxan.jakon.webui

import org.scalajs.dom
import org.scalajs.dom.{Event, HTMLSelectElement, Headers, HttpMethod, RequestInit, RequestMode, RequestRedirect, Response, document}

import scala.concurrent.Future
import scala.scalajs.js
import scala.scalajs.js.{JSON, Promise}

object Ajax {

	import scala.concurrent.ExecutionContext.Implicits.global
	import js.Thenable.Implicits._

	def post(url: String, data: js.Any): Future[String] = {
		 val res = dom.fetch(url, new RequestInit {
			body = js.JSON.stringify(data)
			headers = new Headers(
				js.Array(
					js.Array("Content-Type", "application/json")
				)
			)

			method = HttpMethod.POST
			mode = RequestMode.cors // no-cors, cors, *same-origin
			redirect =  RequestRedirect.follow  // manual, *follow, error
			//referrer=  'no-referrer' // *client, no-referrer
		}).toFuture.flatMap((response: Response) => {
			if (response.ok) {
				response.text().toFuture
			} else {
				println(s"Ajax request failed")
				Promise.reject(response.status.toString).toFuture
			}
		})
		res
	}

}
